cpp/dolfinx/la/SparsityPattern.h

```diff
@@ -70,6 +70,11 @@ class SparsityPattern
   /// Move assignment
   SparsityPattern& operator=(SparsityPattern&& pattern) = default;
 
+  /// Take a SparsityPattern with a block size greater than 1, and
+  /// expand the entries, repeating rows and columns in each block.
+  /// @return SparsityPattern with bs={1, 1}
+  SparsityPattern expand();
```

Comment:
Can this be a free function? Would be better if it's not a class member.

What are preconditions? Should the pattern be finalised first? Is there a reason why the function isn't `const`?

Reply:
Yes, maybe better as a free function. Then I don't need to worry about being `const` 😄